pantalaimon/daemon.py

```diff
@@ -851,9 +853,47 @@ async def send_message(self, request):
 
         # The room isn't encrypted just forward the message.
         if not encrypt:
+            content_msgtype = ""
+            msgtype = request.match_info["event_type"]
+            if content_msgtype == "m.image" or content_msgtype == "m.video" \
+                or content_msgtype == "m.audio" or content_msgtype == "m.file":
+                content_uri = request.match_info["content"]["url"]
+
+                upload = self.store.load_upload(content_uri)
+                if upload is None:
+                    return await self.forward_to_web(request, token=client.access_token)
+
+                mxc = urlparse(content_uri)
+                server_name = mxc.netloc.strip("/")
+                media_id = mxc.path.strip("/")
+                file_name = request.match_info.get("file_name")
+
+                response, decrypted_file, error = await self._load_media(server_name, media_id, file_name, request)
+
+                if response is None and decrypted_file is None and error is None:
+                    return await self.forward_to_web(request, token=client.access_token)
+                if error is ClientConnectionError:
+                    return await self.forward_to_web(request, token=client.access_token)
+                if error is KeyError:
+                    return await self.forward_to_web(request, token=client.access_token)
+
+                if not isinstance(response, DownloadResponse):
```

Comment:
If it's not a `DownloadResponse` it's an error response, again I think returning an error to the client instead of forwarding the request makes more sense.